### PR TITLE
CCUBE-1494

### PR DESCRIPTION
**Pull request by adenteo**
**Changes**
- Resolve circular dependencies from validation styles
- Add contact-specific features
- Render different alert messages
- Add test case for default country code
- Extract validation test case for contact field
- Resolve circular dependency

**Additional information**

-   You may refer to this [ticket](https://jira.ship.gov.sg/browse/CCUBE-1494)

### CODE REVIEW

File: src/components/element-editor/basic-details/elements/contact/default-country-code.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                name="defaultCountryCode"
+                control={control}
+                render={({ field }) => {
+                    const { ref, ...fieldWithoutRef } = field;
+                    return (
+                        <Form.Select
+                            {...fieldWithoutRef}
+                            data-testid="default-country-code"
+                            label="Default country code (optional)"
+                            options={CountryData}
```
Comment (weili-govtech) [inline src/components/element-editor/basic-details/elements/contact/default-country-code.tsx]: need the empty option `No default country code` at the top

File: src/components/element-editor/basic-details/elements/contact/default-country-code.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                            )}
+                            valueExtractor={(item) => item[3]}
+                            listExtractor={(item) => ({
+                                title: item[0] as string,
+                                secondaryLabel: `+${item[3] as string}`,
+                            })}
+                            displayValueExtractor={(item) =>
+                                `${item[0]} (+${item[3]})`
+                            }
+                            onSelectOption={(item) => field.onChange(item[3])}
```
Comment (weili-govtech) [inline src/components/element-editor/basic-details/elements/contact/default-country-code.tsx:66]: should clear the validation rule upon switching between Singapore and other country numbers

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/components/element-editor/basic-details/elements/contact/default-country-code.tsx (hunk 2 above)
Comment (adenteo) [reply src/components/element-editor/basic-details/elements/contact/default-country-code.tsx:66]: Upon switching to other countries, validation rule would unmount and unregister its current values already. I have also set up a `useEffect` in `contact-validation-rule` (in the next branch) to set the value to the default Singapore validation upon first render. Does that logic sound ok?

Comment (weili-govtech) [reply src/components/element-editor/basic-details/elements/contact/default-country-code.tsx:66]: yup sounds ok, as long as this is addressed.

**Pull request merged**